### DISCUSSION TITLE
how can I hear while recording?

### DISCUSSION THREAD

**Discussion by ghost** [ExistentialAudio/BlackHole#133] [General]
it's not clear how to hear while recording. If I set blackhole as output audio I cannot hear what I record.
I understand it has to do with the midi setup but I am not familiar with it and setting some aggregated output didn't work for me. Do I need to set my headphones output channel as the input channel to blackhole? how do I do that?

**Comment by jonasjelonek** [comment]
If you understand you right, you want to record your system audio and also play via headphones. For this, you have to create a Multi-Output-Device with both BlackHole and your regular output (take care of the tip to also select the built-in output but mute it). Then select this MOD as output, BlackHole as input and you can record what you hear.

If you meant something different, please tell us.

**Comment by blacs30** [comment]
Since the question is not absolutely clear: Is it possible to use a microphone as input for recording and hear myself talk at the same time? So in that case it's not about system audio but input from external sources as throughput to the headphones.

**Comment by devinroth** [comment]
I use Logic Pro X for that. Can’t do it without an additional application.

**Comment by blacs30** [comment]
> I use Logic Pro X for that. Can’t do it without an additional application.

Yes, that's one possibility. 
Would be great though if blackhole could also route between audio devices without depending on applications to be open or in record mode.

**Comment by frozenpandaman** [comment]
@blacs30 I'm a fan of this very-lightweight audio routing utility: https://www.macupdate.com/app/mac/11333/linein

**Comment by blacs30** [comment]
@frozenpandaman thanks for sharing. Unfortunately I cannot start it on macOS Catalina :(

**Comment by alreadydone** [comment]
Has anyone encountered this problem: I created a multi-output device consisting of solely "Blackhole 16ch", but once I select it as the output device, any video currently playing immediately halts (in QuickTime it's paused and in YouTube it shows the rotating annulus as if it's buffering). However, using "Blackhole 16ch" both as the output device and as the Microphone in QuickTime, I am indeed able to record videos with sound without problem. My system is Catalina 10.15.3.

**Comment by devinroth** [comment]
Please read the wiki. 

Devin

**Comment by alreadydone** [comment]
Thanks! The relevant entry is https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device#audio-and-video-wont-play-when-using-a-multioutput-device and that works perfectly. I did include both devices in my first try instead of a single device, but didn't pay attention to the order.

> Please read the wiki. Devin
> […](#)
> On May 30, 2020, at 7:55 PM, Junyan Xu ***@***.***> wrote: ﻿ Has anyone encountered this problem: I create a multi-output device consisting of "Blackhole 16ch" only and when I select it as the multi-output device, any video currently playing immediately halts (in QuickTime it's paused and in YouTube it shows the rotating annulus as if it's buffering). However, using "Blackhole 16ch" as both the output device and as the Microphone in QuickTime, I am indeed able to record videos with sound without problem. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.imi.
